extra/interface/ap/src/main/java/org/spongepowered/configurate/interfaces/processor/ConfigImplementationGeneratorProcessor.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Configurate
+ * Copyright (C) zml and Configurate contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spongepowered.configurate.interfaces.processor;
+
+import static org.spongepowered.configurate.interfaces.processor.Utils.isNestedConfig;
+
+import com.google.auto.service.AutoService;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.TypeSpec;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+import org.spongepowered.configurate.interfaces.Constants;
+import org.spongepowered.configurate.objectmapping.ConfigSerializable;
+
+/**
+ * Generates an implementation for a given interface based config,
+ * which then can be read by Configurate.
+ *
+ * @since 4.2.0
+ */
+@AutoService(Processor.class)
+public final class ConfigImplementationGeneratorProcessor extends AbstractProcessor {
```

Comment:
This should be modified to meet the requirements of Gradle's [incremental annotation processing](https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing) (which will likely involve creating two processors, one which creats the interface implementations with some sort of `@GeneratedFrom(SrcClass.class)` annotation, and a second which generates the manifest of interfaces based on that `@GeneratedFrom` annotation).

Reply:
If I read the documentation correctly the `aggregating` type should support this with no further changes (except from the META-INF file)